docs/digital_analog_qc/rydberg-hea.md

```diff
@@ -0,0 +1,106 @@
+Qadence offers simplifies the execution of digital-analog
+workloads on neutral atom quantum computers where the local
+addressability is restricted.
+
+In this regime, which we will refer to as *semi-local addressing*,
+the full Hamiltonian of the qubit system realized with neutral
+atoms comprises the following terms:
+
+$$
+\mathcal{H} = \mathcal{H}_{\textrm{global}} + \mathcal{H}_{\textrm{int}} + \mathcal{H}_{\textrm{local}}
+$$
+
+The first two terms are the standard components of a neutral atom Hamiltonians
+and read as follows:
+
+$$
+    \mathcal{H}_{\textrm{global}} = \frac{\Omega}{2}\sum_{i}^N \left(
+        \textrm{cos}(\phi)\sigma^x_i - \textrm{sin}(\phi)\sigma^y_i \right) -
+        \delta \sum_{i}^N \hat{n}_i \\
+    \mathcal{H}_{\textrm{int}} = \sum_{i<j} \frac{C_6}{|R_i - R_j|^6} \hat{n}_i \hat{n}_j
+$$
+where $\Omega$ is the Rabi frequency, $\phi$ the global phase, $\delta$ the
+detuning which can all be time-dependent (here omitted for simplicity). The operator
+$\hat{n}_i = \frac{1+\sigma_i^z}{2}$ is the occupation operator for the Rydberg state.
+
+The local addressability term read instead as follows:
+
+$$
+    \mathcal{H}_{\textrm{local}}(w^{drv}, w^{det}) = \frac{\tilde{\Omega}}{2}\sum_{i}^N
+        w_i^{drv}\left(\textrm{cos}(\phi)\sigma^x_i - \textrm{sin}(\phi)\sigma^y_i \right) -
+        \Delta \sum_{i}^N w_i^{det} \hat{n}_i
+$$
+
+In this Hamiltonian, the local addressing pattern in both Rabi frequency and detuning
+is determined by the weights $w^{drv} = \{w_i^{drv}\}$ and $w^{det} = \{w_i^{det}\}$ respectively. These
+weights are assigned before starting the simulation and they should have a unit sum.
+Their action is to effectively modulate the amplitude of the local drive/detuning pulses given by
+$\tilde{\Omega}$ and $\Delta$ which are here considered time-independent for simplicity.
+
+Qadence implements the Hamiltonian above in two different flavors of increasing complexity described
+below.
+
+## Circuit constructor
+
+The [`rydberg_hea`][qadence.constructors.rydberg_hea.rydberg_hea] allows to
+build a circuit instance implementing a basic version of the Hamiltonian
+evolution described above where both $\Delta$ and $\tilde{\Omega}$ coefficients
+are considered constants and **no global drive and detuning** are allowed. Furthermore,
+this implementation does not perform any checks on the weights normalization, thus
+making it not realistic.
+
+You can easily create a Rydberg hardware efficient ansatz implementing multiple layers
+of the evolution generated by the local addressing Hamiltonian:
+
+$$
+\mathcal{H}_{evo} = \sum_j \mathcal{H}_{\textrm{local}}(w_{j}^{drv}, w_{j}^{det})
+$$
+
+Notice that in real-device implementation, one layer only is usually possible.
+
+```python exec="on" source="material-block" result="json" session="rydb-hea"
+import qadence as qd
+from qadence import rydberg_hea
+
+n_qubits = 4
+n_layers = 2
+
+# determine the interaction
+register = qd.Register.line(n_qubits)
```

Comment:
afaik, this line does not directly determine the interaction, but rather the atom layout. The interaction cutoffs (NN, NNN etc) can be chosen independently from this

Reply:
Yes, good point. I wanted to say something like: "determine the interaction strength" but I could not find the right wording. I will change it.